lib/kramdown/parser/gfm.rb

```diff
@@ -112,22 +112,16 @@ def generate_gfm_header_id(text)
         @options[:auto_id_prefix] + result
       end
 
-      ATX_HEADER_START = /^\#{1,6}\s/
+      ATX_HEADER_START = /^(?<level>\#{1,6})[\t ]+(?<contents>.*)\n/
       define_parser(:atx_header_gfm, ATX_HEADER_START, nil, 'parse_atx_header')
       define_parser(:atx_header_gfm_quirk, ATX_HEADER_START)
 
       # Copied from kramdown/parser/kramdown/header.rb, removed the first line
       def parse_atx_header_gfm_quirk
-        start_line_number = @src.current_line_number
-        @src.check(ATX_HEADER_MATCH)
-        level, text, id = @src[1], @src[2].to_s.strip, @src[3]
+        text, id = parse_header_contents
+        text.sub!(/[\t ]#+\z/, '') && text.rstrip!
```

Comment:
Is there a reason why these two statements are on one line? I would rather prefer them on separate lines.

Reply:
There is no need to call `rstrip!` if `sub!` return nil.

The other options here are:

A:

```
text.rstrip! if text.sub!(...)
```

B:

```
if text.sub!(...)
  text.rstrip!
end
```

I think the current one (`&&`) is quite readable but can go with `A` or `B` if that's what you prefer.

Reply:
Personally I would just put them on separate lines and take this minimal performance hit because it would be easier to see what's going on without the additional conditional.